cmd/pulumi-test-language/tests.go

```diff
@@ -1163,6 +1163,57 @@ var languageTests = map[string]languageTest{
 			},
 		},
 	},
+	"l2-invoke-options-depends-on": {
+		providers: []plugin.Provider{&providers.SimpleInvokeProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					requireStackResource(l, err, changes)
+					require.Len(l, snap.Resources, 5, "expected 5 resources")
+					// TODO: the root stack must be the first resource to be registered
+					// such that snap.Resources[0].Type == resource.RootStackType
+					// however with the python SDK, that is not the case, instead the default
+					// provider gets registered first. This is indicating that something might be wrong
+					// with the how python SDK registers resources
```

Comment:
Worth raising an issue for this so that we can track/remove this TODO when it's resolved?

Reply:
Yep! I copied this from the other invoke tests, but we should definitely have a tracking issue for this.

I created the issue https://github.com/pulumi/pulumi/issues/17816 and I'll reference it from the comments.